dodo.py

```diff
@@ -81,20 +81,81 @@ def task_openspiel_compile():
     }
 
 
-def task_forecast():
+#####################
+# FORECASTING TASKS #
+#####################
+
+FORECAST_QUERY_LOG_DIR = Path("./forecast/data/extracted/extended/")
+
+FORECAST_ARTIFACTS_DIR = Path("artifacts")
+FORECAST_PREPROCESSOR_ARTIFACT = FORECAST_ARTIFACTS_DIR.joinpath(
+    "preprocessed.parquet.gzip"
+)
+FORECAST_CLUSTER_ARTIFACT = FORECAST_ARTIFACTS_DIR.joinpath("clustered.parquet")
+FORECAST_MODEL_DIR = FORECAST_ARTIFACTS_DIR.joinpath("models")
+FORECAST_PREDICTION_CSV = FORECAST_ARTIFACTS_DIR.joinpath("forecast.csv")
+
+FORECAST_PREDICTION_START = "2021-12-06 14:24:32 EST"
+FORECAST_PREDICTION_END = "2021-12-06 14:24:36 EST"
+
+
+def task_forecast_preprocess():
     """
-    Forecast: forecast the workload.
+    Forecast Preprocess: create templatized queries data for ingest into
+    forecaster training or prediction
     """
+
+    preprocessor_action = (
+        "python3 ./forecast/preprocessor.py"
+        f" --query-log-folder {FORECAST_QUERY_LOG_DIR.absolute()} "
+        f"--output-parquet {FORECAST_PREPROCESSOR_ARTIFACT.absolute()}"
+    )
+
     return {
         "actions": [
-            "mkdir -p artifacts",
+            f"mkdir -p {FORECAST_ARTIFACTS_DIR.absolute()}",
```

Comment:
`.absolute()` returns the absolute path _as of the time of the call_.
This can lead to unintuitive interactions with chdir, e.g.,

```
>>> from pathlib import Path
>>> from os import chdir
>>> capybaraland = Path('./capybaraland')
>>> str(capybaraland.absolute())
'/home/kapi/capybaraland'
>>> chdir('capybaraland')
>>> str(capybaraland.absolute())
'/home/kapi/capybaraland/capybaraland'
```

I would put all calls to absolute up there in the original path definition.